src/Plugin/Field/FieldWidget/ModeNumberWidget.php

```diff
@@ -0,0 +1,248 @@
+<?php
+
+namespace Drupal\commerce\Plugin\Field\FieldWidget;
+
+use Drupal\Core\Field\FieldFilteredMarkup;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Field\WidgetBase;
+use Drupal\Core\Form\FormStateInterface;
+use Symfony\Component\Validator\ConstraintViolationInterface;
+use Drupal\Core\Field\Plugin\Field\FieldWidget\NumberWidget;
+
+/**
+ * Plugin implementation of the 'mode_number' widget.
+ *
+ * @FieldWidget(
+ *   id = "mode_number",
+ *   label = @Translation("Mode number field"),
+ *   field_types = {
+ *     "integer",
+ *     "decimal",
+ *     "float"
+ *   }
+ * )
+ */
+class ModeNumberWidget extends NumberWidget {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultSettings() {
+    return array(
+      'placeholder' => NULL,
```

Comment:
Well, looked into the test once again and all that I've found is this:

src/Plugin/Field/FieldWidget/ModeNumberWidget.php

Line 7
unused use statement need to be removed.

Line 62, 67
leading white space in descriptions need to be removed.

Line 74, 115, 122

t() wrapper for dynamic string values need to be removed.

Could you point where Travis complains about undefined default values?